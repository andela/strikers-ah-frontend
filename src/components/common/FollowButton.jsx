/* eslint-disable prefer-destructuring */
/* eslint-disable react/destructuring-assignment */
import React, { Component } from 'react';
import { connect } from 'react-redux';
// import { Redirect } from 'react-router-dom';
import '../../styles/css/followBtn.css';
import {
  followUser,
  unfollowUser,
  checkProfileFollowStatus,
} from '../../redux/actions/followingAction';
/**
 * Follow button
 */
class FollowButton extends Component {
  /**
   * @author frank harerimana
   * @param {*} props
   * @returns {*} button
   */
  constructor(props) {
    super(props);
    this.state = {
      username: 'frank',
    };
    this.handleClick = this.handleClick.bind(this);
  }

  /**
   * @author frank harerimana
   * @param {*} event
   * @returns {*} button
   */
  componentDidMount() {
    const token = localStorage.getItem('token');
    if (token !== null) {
      this.checkStatus();
    }
  }

  /**
   * @author frank harerimana
   * @param {*} event
   * @returns {*} button
   */
  checkUsername() {
    // let username;
    // if (this.props.Article.article.authorid != null) {
    //   username = this.props.articleAuthor.article[0].user.username;
    // } else if (this.props.userProfile != null) {
    // const username = this.props.userProfile.user.profile.username;
    // }
    return this.state.username;
  }

  /**
   * @author frank harerimana
   * @param {*} event
   * @returns {*} button
   */
  checkStatus() {
    const username = this.checkUsername();
    this.props.checkProfileFollowStatus(username);
  }

  /**
   * @author frank harerimana
   * @param {*} event
   * @returns {*} button
   */
  handleClick(event) {
    event.preventDefault();
    if (localStorage.getItem('token') != null) {
      const { status } = this.props.following[0];
      const username = this.checkUsername();
      if (status === 'follow') {
        this.props.followUser(username);
      } else {
        this.props.unfollowUser(username);
      }
    } else {
      // eslint-disable-next-line no-return-assign
      return (window.location.href = '/login');
    }
  }

  /**
   * @author frank harerimana
   * @param {*} props
   * @returns {*} button
   */
  render() {
    const { status } = this.props.following[0];
    return (
      // <div className="follow-btn-div">
      <button
        type="button"
        id={status === 'following' ? 'following-btn' : 'follow-btn'}
        className={this.props.className}
        onClick={this.handleClick}
      >
        {status}
      </button>
      // </div>
    );
  }
}
const mapStateToProps = state => ({
  following: state.following,
  userProfile: state.userProfile,
  Article: state.Article,
});

export default connect(
  mapStateToProps,
  { followUser, unfollowUser, checkProfileFollowStatus },
)(FollowButton);
