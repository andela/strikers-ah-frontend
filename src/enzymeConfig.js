import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import 'jest-localstorage-mock';

export default Enzyme.configure({
  adapter: new Adapter(),
});
