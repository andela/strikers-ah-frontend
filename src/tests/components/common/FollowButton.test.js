import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import '../../../enzymeConfig';
import FollowButton from '../../../components/common/FollowButton';

const mockStore = configureStore();
const initialState = [
  {
    status: 'follow',
  },
];
const props = {
  following: {
    0: {
      status: 'following',
    },
  },
  handleClick: jest.fn(),
};
let store, wrapper;

beforeEach(() => {
  store = mockStore(initialState);
  wrapper = shallow(<FollowButton store={store} {...props} />);
});
describe('`FollowButton.jsx`', () => {
  test('should render', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
