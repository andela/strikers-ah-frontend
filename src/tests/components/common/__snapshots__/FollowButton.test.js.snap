// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`\`FollowButton.jsx\` should render 1`] = `ShallowWrapper {}`;
